### DISCUSSION TITLE
Using custom JSX transforms in Deno?

### DISCUSSION THREAD

**Discussion by trusktr** [denoland/deno#12261] [Q&A]
How does one use a custom JSX transform in Deno?

For example, I need this:

```js
const div = <div>hello</div>
```

to be essentially this at runtime:

```js
const div = document.createElement('div')
div.textContent = 'hello'
```

In non-Deno projects, this can be achieved with https://www.npmjs.com/package/babel-preset-solid for Babel, for example.

**Comment by lucacasonato** [comment]
A custom JSX factory function can be specified with the `// @jsx` and `// @jsxFrag` pragmas. There is no way to do custom AST manipulation for the JSX syntax.
